src/BccCode.Linq/Client/QueryProvider/ApiQueryProvider.cs

```diff
@@ -385,7 +387,32 @@ protected override Expression VisitMember(MemberExpression node)
         {
             if (_activeParameters.TryGetValue(parameterNode, out var apiCaller))
             {
-                string ApiMemberName() => node.Member.Name.ToCamelCase();
+                string ApiMemberToJsonString(MemberInfo member)
+                {
+                    var dataMemberAttribute = member.GetCustomAttribute<DataMemberAttribute>();
+                    if (dataMemberAttribute != null)
+                    {
+                        return dataMemberAttribute.Name;
+                    }
+                    
+#if NET6_0_OR_GREATER
+                    var jsonPropertyNameAttribute = member.GetCustomAttribute<System.Text.Json.Serialization.JsonPropertyNameAttribute>();
+                    if (jsonPropertyNameAttribute != null)
+                    {
+                        return jsonPropertyNameAttribute.Name;
+                    }
+#endif
+
+                    var jsonPropertyAttribute = member.GetCustomAttribute<JsonPropertyAttribute>();
+                    if (jsonPropertyAttribute != null)
+                    {
+                        return jsonPropertyAttribute.PropertyName ?? member.Name.ToCamelCase();
```

Comment:
Should .ToCamelCase() really be applied here?

Reply:
This is the fallback when the `JsonPropertyAttribute` is used, but `PropertyName` is `null`. This could be the case when you want to not specify the name with the `JsonPropertyAttribute`, but do something else / like defining the order of the properties. I just added it because the `PropertyName` is nullable.
This does not apply to `JsonPropertyNameAttribute` (new C# json parser from Microsoft)

Reply:
@rvanoord ping